gravitee-apim-rest-api/gravitee-apim-rest-api-portal/gravitee-apim-rest-api-portal-rest/src/test/java/io/gravitee/rest/api/portal/rest/resource/SubscriptionResourceTest.java

```diff
@@ -73,285 +76,298 @@ public void init() {
+            verify(subscriptionService).update(eq(GraviteeContext.getExecutionContext()), subscriptionCaptor.capture());
+            assertEquals(SUBSCRIPTION, subscriptionCaptor.getValue().getSubscriptionId());
+        }
     }
 
-    @Getter
-    @AllArgsConstructor
-    private class SubscriptionConfiguration {
+    @Nested
+    class ChangeSubscriptionConsumerStatus {
+
+        @Test
+        void shouldPauseSubscriptionByConsumer() {
+            Response response = target(SUBSCRIPTION).path("_changeConsumerStatus").queryParam("status", "STOPPED").request().post(null);
+
+            assertEquals(200, response.getStatus());
+            verify(subscriptionService, times(1)).pauseConsumer(GraviteeContext.getExecutionContext(), subscriptionEntity.getId());
+        }
+
+        @Test
+        void shouldResumeSubscriptionByConsumer() {
+            Response response = target(SUBSCRIPTION).path("_changeConsumerStatus").queryParam("status", "STARTED").request().post(null);
+
+            assertEquals(200, response.getStatus());
+            verify(subscriptionService, times(1)).resumeConsumer(GraviteeContext.getExecutionContext(), subscriptionEntity.getId());
+        }
+
+        @Test
+        void shouldHaveBadRequestIfTryingAWrongConsumerStatus() {
+            Response response = target(SUBSCRIPTION).path("_changeConsumerStatus").queryParam("status", "INVALID").request().post(null);
+
+            assertEquals(400, response.getStatus());
+            verify(subscriptionService, times(0)).pauseConsumer(GraviteeContext.getExecutionContext(), subscriptionEntity.getId());
+            verify(subscriptionService, times(0)).resumeConsumer(GraviteeContext.getExecutionContext(), subscriptionEntity.getId());
+        }
+
+        @Test
+        void shouldBeForbiddenWhenUpdatingConsumerStatus() {
+            reset(permissionService);
+
+            doReturn(false)
+                .when(permissionService)
+                .hasPermission(
+                    GraviteeContext.getExecutionContext(),
+                    RolePermission.APPLICATION_SUBSCRIPTION,
+                    APPLICATION,
+                    RolePermissionAction.UPDATE
+                );
+            Response response = target(SUBSCRIPTION).path("_changeConsumerStatus").queryParam("status", "STOPPED").request().post(null);
+
+            assertEquals(HttpStatusCode.FORBIDDEN_403, response.getStatus());
+        }
+    }
 
-        private String url;
+    @Nested
+    class ResumeFailedSubscription {
+
+        @Test
+        void shouldReturn403IfIncorrectPermission() {
+            reset(permissionService);
+
+            doReturn(false)
+                .when(permissionService)
+                .hasPermission(
+                    GraviteeContext.getExecutionContext(),
+                    RolePermission.APPLICATION_SUBSCRIPTION,
+                    APPLICATION,
+                    RolePermissionAction.UPDATE
+                );
```

Comment:
At least for new tests can you use

```suggestion
            when(permissionService.hasPermission(
                    GraviteeContext.getExecutionContext(),
                    RolePermission.APPLICATION_SUBSCRIPTION,
                    APPLICATION,
                    RolePermissionAction.UPDATE
                )
                .thenRetun(false);
```

this have 2 advantages:

- type safe: if the type returned by hasPermission() change, it’s reflected here
- the shape when/thenReturn put mockito in strict mode and throw an error if the fake is never call. This improve the maintanability of test (for case where it’s a problem we can use lenient)